submissions/vakulinina/friends-app/main.js

```diff
@@ -0,0 +1,178 @@
+  return a.name.first.toLowerCase() <= b.name.first.toLowerCase() ? -1 : 1
+};
+
+const compareByAge = (a, b) => {
+  return a.dob.age - b.dob.age
+};
+
+const sortByName = users => {
+  if (filter.nameSort === "asc") {
+    return users.sort((a, b) => compareByName(a, b))
+  } else {
+    return users.sort((a, b) => compareByName(b, a))
+  }
+};
+
+const sortByAge = users => {
+  if (filter.ageSort === "asc") {
+    return users.sort((a, b) => compareByAge(a, b))
+  } else {
+    return users.sort((a, b) => compareByAge(b, a))
+  }
+};
+
+const filterByAge = users => {
+  const [ageFrom, ageTo] = [filter.ageFrom, filter.ageTo].sort((a, b) => a - b);
+  return users.filter(({ dob: { age } }) => (age >= ageFrom && age <= ageTo))
+};
+
+const filterByGender = users => {
+  return users.filter(user => user.gender === filter.gender)
+};
+
+const doSearch = users => {
+  return users.filter(({ name: { first, last }, location: { city } }) => {
+    const fullName = `${first.toLowerCase()} ${last.toLowerCase()}`;
+    return fullName.includes(filter.searchValue) || city.toLowerCase().includes(filter.searchValue)
+  })
+};
+
+const filterUsers = () => {
+  let filteredUsers = [...allUsers];
+  if (filter.ageSort) {
+    filteredUsers = sortByAge(filteredUsers);
+  }
+  if (filter.nameSort) {
+    filteredUsers = sortByName(filteredUsers);
+  }
+  if (filter.ageFrom || filter.ageTo) {
+    filteredUsers = filterByAge(filteredUsers);
+  }
+  if (filter.gender !== 'any') {
+    filteredUsers = filterByGender(filteredUsers);
+  }
+  if (filter.searchValue) {
+    filteredUsers = doSearch(filteredUsers);
+  }
+  renderUsers(filteredUsers)
+};
+
+const setNameSortType = () => {
+  if (filter.nameSort === 'asc') {
+    filter.nameSort = 'desc';
+    sortByNameButton.classList.remove('asc')
+    sortByNameButton.classList.add('desc')
+  } else {
+    filter.nameSort = 'asc';
+    sortByNameButton.classList.remove('desc')
+    sortByNameButton.classList.add('asc')
+  }
+  sortByAgeButton.classList.remove('asc', 'desc')
+  filter.ageSort = null;
+};
+
+const setAgeSortType = () => {
+  if (filter.ageSort === 'asc') {
+    filter.ageSort = 'desc';
+    sortByAgeButton.classList.remove('asc')
+    sortByAgeButton.classList.add('desc')
+  } else {
+    filter.ageSort = 'asc';
+    sortByAgeButton.classList.remove('desc')
+    sortByAgeButton.classList.add('asc')
+  }
+  sortByNameButton.classList.remove('asc', 'desc')
+  filter.nameSort = null;
+};
+
+const handleInput = ({ target }) => {
+  switch (target.id) {
+    case 'name-sort':
+      setNameSortType()
+      break;
+    case 'age-sort':
+      setAgeSortType()
+      break;
+    case 'age-from':
+      filter.ageFrom = target.value || MIN_AGE;
+      break;
+    case 'age-to':
+      filter.ageTo = target.value || MAX_AGE;
+      break;
+    case 'search-field':
+      filter.searchValue = target.value.toLowerCase();
+      break;
+    default:
+      filter.gender = target.value;
+  }
```

Comment:
Try to replace `switch` with map.

Something like https://github.com/kottans/frontend-2021-homeworks/pull/434#discussion_r576145474